addons/purchase_requisition/tests/test_purchase_requisition.py

```diff
@@ -348,3 +348,66 @@ def test_11_alternative_po_from_po_with_requisition_id(self):
 
         po_2 = po_1.alternative_po_ids - po_1
         self.assertFalse(po_2.requisition_id, "The requisition_id should not be set in the alternative purchase order")
+
+    def test_12_alternative_po_line_different_currency(self):
+        """ Check alternative PO with different currency is compared correctly"""
+        vendor_eur = self.env['res.partner'].create({
+            'name': 'eur',
+        })
+        vendor_usd = self.env['res.partner'].create({
+            'name': 'eur',
+        })
+
+        currency_eur = self.env.ref("base.EUR")
+        currency_usd = self.env.ref("base.USD")
+
+        # 2 EUR = 1 USD
+        self.env['res.currency.rate'].create({
+            'currency_id': self.env.ref('base.USD').id,
+            'rate': 1
+        })
+
+        self.env['res.currency.rate'].create({
+            'currency_id': self.env.ref('base.EUR').id,
+            'rate': 2
+        })
```

Comment:
you can batch create these 2 at the same time or drop the USD rate since the test default is already USD and it shouldn't convert it (i.e. this rate won't be used)